routers/api/packages/api.go

```diff
@@ -135,6 +136,49 @@ func CommonRoutes() *web.Router {
 				})
 			})
 		}, reqPackageAccess(perm.AccessModeRead))
+		r.Group("/arch", func() {
+			r.Methods("HEAD,GET", "/repository.key", arch.GetRepositoryKey)
+
+			r.Methods("HEAD,GET,PUT,DELETE", "*", func(ctx *context.Context) {
+				path := strings.Trim(ctx.PathParam("*"), "/")
+
+				if ctx.Req.Method == "PUT" {
+					reqPackageAccess(perm.AccessModeWrite)(ctx)
+					if ctx.Written() {
+						return
+					}
+					ctx.SetPathParam("repository", path)
+					arch.UploadPackageFile(ctx)
+					return
+				}
+
+				pathFields := strings.Split(path, "/")
+				pathFieldsLen := len(pathFields)
+
+				if (ctx.Req.Method == "HEAD" || ctx.Req.Method == "GET") && pathFieldsLen >= 2 {
+					ctx.SetPathParam("repository", strings.Join(pathFields[:pathFieldsLen-2], "/"))
+					ctx.SetPathParam("architecture", pathFields[pathFieldsLen-2])
+					ctx.SetPathParam("filename", pathFields[pathFieldsLen-1])
+					arch.GetPackageOrRepositoryFile(ctx)
+					return
+				}
+
+				if ctx.Req.Method == "DELETE" && pathFieldsLen >= 3 {
+					reqPackageAccess(perm.AccessModeWrite)(ctx)
+					if ctx.Written() {
+						return
+					}
+					ctx.SetPathParam("repository", strings.Join(pathFields[:pathFieldsLen-3], "/"))
+					ctx.SetPathParam("architecture", pathFields[pathFieldsLen-3])
```

Comment:
Why use `<architecture>/<name>/<version>` instead of `<name>/<version>/<architecture>`, it looks inconsistent with other package styles.

Reply:
Changed it but in retrospect I would have changed the order for Debian and rpm because it should be sorted by frequency of the underlaying items.
group/repo > architecture > package > version